src/components/Form/Form.js

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import './Form.css';
+
+const Form = ({ users, onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [person, setPerson] = useState({});
+  const [validTitle, setValidTitle] = useState(true);
+  const [validPerson, setValidPerson] = useState(true);
+
+  const addTask = (event) => {
+    event.preventDefault();
+
+    if (title === '') {
+      setValidTitle(false);
+    }
+
+    if (!person.name) {
+      setValidPerson(false);
+    }
+
+    if (title && person.name) {
+      const todo = {
+        title,
+        userId: person.id,
+        user: person,
+        completed: false,
+      };
+
+      onAdd(todo);
+      setTitle('');
+      setPerson({});
+    }
+  };
+
+  const changeTitle = (event) => {
+    setTitle(event.target.value);
+    setValidTitle(true);
+  };
+
+  const changeUser = (event) => {
+    setPerson(JSON.parse(event.target.value));
+    setValidPerson(true);
+  };
```

Comment:
no need to parse anything, just use array.find method on users to find a user by name or id